keps/sig-scheduling/2891-simplified-config/README.md

```diff
@@ -0,0 +1,861 @@
+implementation within the scheduling framework to use different extension points.
+
+#### Story 2
+
+As a user, I wish to enable plugin `XYZ` for only *some* of its applicable extension 
+points (ie `Filter` but *not* `Score`). I still do not care about extension points such 
+as `PreFilter`. I want to have control over which subsets of extension points are enabled.
+
+#### Story 3
+
+As a plugin developer, I want users to be able to easily enable my plugin for all applicable 
+extension points (or a subset of extension points as defined either by me or my users' needs). 
+I want this available with minimal code updates required from me.
+
+### Notes/Constraints/Caveats (Optional)
+
+* changes should have minimal effect on existing plugins and users
+* 
+
+### Risks and Mitigations
+
+* out-of-tree plugins may be developed by anyone, so the information that is 
+passed to them and received from them should be kept to a minimum. However these are 
+unsupported so security is ultimately up to users to be aware of.
+* UX will need to be carefully examined, as a new config field can introduce confusing 
+and unexpected outcomes if different combinations of options are not well-defined.
+
+## Design Details
+
+An example of the new user-facing profile config would be:
+
+```yaml
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+profiles:
+  - schedulerName: default-scheduler
+    plugins:
+      multiPoint:
+        enabled:
+        - name: "PodTopologySpread" ## PreFilter/Filter/PreScore/Score
+          weight: 3
+        - name: "VolumeBinding" ## PreFilter/Filter/Reserve/PreBind
+        - name: "PrioritySort" ## QueueSort
+```
+
+Specific sections (eg, `Score`) could still be entirely disabled like so:
+
+```yaml
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+profiles:
+  - schedulerName: default-scheduler
+    plugins:
+      multiPoint:
+        enabled:
+        - name: "PodTopologySpread" ## PreFilter/Filter
+          weight: 3
+        - name: "VolumeBinding" ## PreFilter/Filter/Reserve/PreBind
+        - name: "PrioritySort" ## QueueSort
+      score:
+        disabled:
+        - "*"
+```
+(Note that in the above, now `PodTopologySpread` will not be registered for `Score` or `PreScore`)
+
+(Also note, the semantics of doing such an approach may vary depending on the implementation design 
+chosen to add multiPoint plugins)
+
+To do this, we must first define a new extension point `MultiPoint` in [`Plugins`](https://github.com/kubernetes/kube-scheduler/blob/bb2d43a/config/v1beta2/types.go#L154-L159)
+```go
+type Plugins struct {
+  MultiPoint PluginSet `json:"multiPoint,omitEmpty"`
+...
+}
+```
+
+From this point there are 2 implementation options. For both, we will need to support the 
+current "regular" extension points at least until they are able to be deprecated.
+
+[**Option 1**](#Option-1)
```

Comment:
Which option are you proposing we take? That one that we don't take should go in the "Alternatives" section.

Reply:
I personally prefer option 1, but I may be biased in that regard. I tried to present them here equally because it sounded like there was a good amount of support for option 2

Reply:
Here lies a divergence of views. We can look at it in two personas: administrators / ops who configure the scheduler, and plugin developers.

For option 1, it enables the plugin developers the most flexibility to configure a plugin programmatically and dynamically - which may resolve some cases we haven't met so far. Also, it's possible to use `multiPoint` as the **only** way to compose scheduler config in the future. For admins, they get more work to do - in terms of configuration as well as understanding the plugin registration logic so as not to be surprised by something like plugin args cascading with enablement of a plugin. Also, this brings a challenge to explain everything precisely in layman's terms.

For option 2, we lost the flexibility of registrating a multiPoint plugin in runtime. So every config fields given by the admins are the final configuration. This also implies it's a must to keep multiPoint and other points in the future v1 config.

Generally, I'm inclined to option 2 as it defines a smaller but workable scope, and less code complexity and semantic ambiguity to maintain. Also, if the user wants to modify the scheduler config, modifying a static config yaml can be way easier than modifying the src code.